Compilation failure on MacOS w. Apple silicon

I get build errors in the last steps around `compiler-rt` on apple M1. Seems `OSX supported arches:` is empty on my setup.

llvm-mos: 97e1882aedf5dba49ae1071c907503036d1c5fc7
cmake: 3.28.1
macos: sonoma 14.3

```console
cmake -C clang/cmake/caches/MOS.cmake -G Ninja -S llvm -B build -DCMAKE_INSTALL_PREFIX=$HOME/llvm-mos
cmake --build build


-- LLVM host triple: arm64-apple-darwin23.3.0
-- LLVM default target triple: mos-unknown-unknown
-- Found libtool - /Users/mikael/github/llvm-mos/build/bin/llvm-libtool-darwin
CMake Deprecation Warning at /Users/mikael/github/llvm-mos/cmake/Modules/CMakePolicy.cmake:6 (cmake_policy):
  The OLD behavior for policy CMP0114 will be removed from a future version
  of CMake.

  The cmake-policies(7) manual explains that the OLD behaviors of all
  policies are deprecated and that a policy should be set to OLD only under
  specific short-term circumstances.  Projects should be ported to the NEW
  behavior and not rely on setting a policy to OLD.
Call Stack (most recent call first):
  /Users/mikael/github/llvm-mos/compiler-rt/CMakeLists.txt:11 (include)


CMake Deprecation Warning at /Users/mikael/github/llvm-mos/cmake/Modules/CMakePolicy.cmake:11 (cmake_policy):
  The OLD behavior for policy CMP0116 will be removed from a future version
  of CMake.

  The cmake-policies(7) manual explains that the OLD behaviors of all
  policies are deprecated and that a policy should be set to OLD only under
  specific short-term circumstances.  Projects should be ported to the NEW
  behavior and not rely on setting a policy to OLD.
Call Stack (most recent call first):
  /Users/mikael/github/llvm-mos/compiler-rt/CMakeLists.txt:11 (include)


-- Found OSX_SYSROOT: /Library/Developer/CommandLineTools/SDKs/MacOSX.sdk
-- Checking DARWIN_iphoneos_SYSROOT - ''
CMake Warning at /Users/mikael/github/llvm-mos/compiler-rt/cmake/base-config-ix.cmake:152 (message):
  iOS SDK not found! Building compiler-rt without iOS support.
Call Stack (most recent call first):
  /Users/mikael/github/llvm-mos/compiler-rt/CMakeLists.txt:36 (include)


-- Found libtool - /Users/mikael/github/llvm-mos/build/bin/llvm-libtool-darwin
-- cmake c compiler target: mos-unknown-unknown
-- Performing Test -ftrivial-auto-var-init=pattern
-- Performing Test -ftrivial-auto-var-init=pattern - Failed
-- Checking DARWIN_iphonesimulator_SYSROOT - ''
-- Checking DARWIN_iphoneos_SYSROOT - ''
-- Checking DARWIN_watchsimulator_SYSROOT - ''
-- Checking DARWIN_watchos_SYSROOT - ''
-- Checking DARWIN_appletvsimulator_SYSROOT - ''
-- Checking DARWIN_appletvos_SYSROOT - ''
-- Got ld supported ARCHES: armv6 armv7 armv7s arm64 arm64e arm64_32 i386 x86_64 x86_64h
-- Toolchain supported arches: armv6;armv7;armv7s;arm64;arm64e;arm64_32;i386;x86_64;x86_64h
-- Finding valid architectures for osx...
-- Disabling i386 slice for DARWIN_osx_ARCHS
-- OSX supported arches:
-- Compiler-RT supported architectures:
CMake Error at /Users/mikael/github/llvm-mos/compiler-rt/CMakeLists.txt:464 (message):
  -g is not supported by host compiler


-- Configuring incomplete, errors occurred!
FAILED: runtimes/runtimes-mos-unknown-unknown-stamps/runtimes-mos-unknown-unknown-configure /Users/mikael/github/llvm-mos/build/runtimes/runtimes-mos-unknown-unknown-stamps/runtimes-mos-unknown-unknown-configure
cd /Users/mikael/github/llvm-mos/build/runtimes/runtimes-mos-unknown-unknown-bins && /opt/homebrew/Cellar/cmake/3.28.1/bin/cmake --no-warn-unused-cli -DCMAKE_C_COMPILER=/Users/mikael/github/llvm-mos/build/./bin/clang -DCMAKE_CXX_COMPILER=/Users/mikael/github/llvm-mos/build/./bin/clang++ -DCMAKE_ASM_COMPILER=/Users/mikael/github/llvm-mos/build/./bin/clang -DCMAKE_AR=/Users/mikael/github/llvm-mos/build/./bin/llvm-ar -DCMAKE_LIBTOOL=/Users/mikael/github/llvm-mos/build/./bin/llvm-libtool-darwin -DCMAKE_LIPO=/Users/mikael/github/llvm-mos/build/./bin/llvm-lipo -DCMAKE_RANLIB=/Users/mikael/github/llvm-mos/build/./bin/llvm-ranlib -DCMAKE_NM=/Users/mikael/github/llvm-mos/build/./bin/llvm-nm -DCMAKE_OBJDUMP=/Users/mikael/github/llvm-mos/build/./bin/llvm-objdump -DCMAKE_C_COMPILER_TARGET=mos-unknown-unknown -DCMAKE_CXX_COMPILER_TARGET=mos-unknown-unknown -DCMAKE_ASM_COMPILER_TARGET=mos-unknown-unknown -DCMAKE_INSTALL_PREFIX=/Users/mikael/llvm-mos -DLLVM_BINARY_DIR=/Users/mikael/github/llvm-mos/build -DLLVM_CONFIG_PATH=/Users/mikael/github/llvm-mos/build/bin/llvm-config -DLLVM_ENABLE_WERROR=OFF -DLLVM_HOST_TRIPLE=arm64-apple-darwin23.3.0 -DLLVM_HAVE_LINK_VERSION_SCRIPT=0 -DLLVM_USE_RELATIVE_PATHS_IN_DEBUG_INFO=OFF -DLLVM_USE_RELATIVE_PATHS_IN_FILES=OFF -DLLVM_LIT_ARGS=-sv -DLLVM_SOURCE_PREFIX= -DPACKAGE_VERSION=19.0.0git -DCMAKE_BUILD_TYPE=MinSizeRel -DCMAKE_MAKE_PROGRAM=/opt/homebrew/bin/ninja -DCMAKE_EXPORT_COMPILE_COMMANDS=1 -DCOMPILER_RT_BUILD_BUILTINS=OFF -DLLVM_INCLUDE_TESTS=ON -DLLVM_ENABLE_PROJECTS_USED=ON -DLLVM_ENABLE_PER_TARGET_RUNTIME_DIR=OFF -DCMAKE_C_COMPILER_WORKS=ON -DCMAKE_CXX_COMPILER_WORKS=ON -DCMAKE_ASM_COMPILER_WORKS=ON -DCOMPILER_RT_DEFAULT_TARGET_ONLY=ON -DLLVM_RUNTIMES_TARGET=mos-unknown-unknown -DHAVE_LLVM_LIT=ON -DCLANG_RESOURCE_DIR= -DLLVM_DEFAULT_TARGET_TRIPLE=mos-unknown-unknown -DLLVM_ENABLE_RUNTIMES=compiler-rt -DLLVM_USE_LINKER= -GNinja -S /Users/mikael/github/llvm-mos/llvm/runtimes/../../runtimes -B /Users/mikael/github/llvm-mos/build/runtimes/runtimes-mos-unknown-unknown-bins && /opt/homebrew/Cellar/cmake/3.28.1/bin/cmake -E touch /Users/mikael/github/llvm-mos/build/runtimes/runtimes-mos-unknown-unknown-stamps/runtimes-mos-unknown-unknown-configure
ninja: build stopped: subcommand failed.
```

Fails for me as well, with the following errors on config:
```
CMake Error at cmake/modules/LLVMDistributionSupport.cmake:276 (message):
  Specified distribution component 'clang-apply-replacements' doesn't have an
  install target
Call Stack (most recent call first):
  CMakeLists.txt:1367 (llvm_distribution_add_targets)


CMake Error at cmake/modules/LLVMDistributionSupport.cmake:284 (message):
  Specified distribution component 'clang-apply-replacements' doesn't have an
  install-stripped target.  Its installation target creation should be
  changed to use add_llvm_install_targets, or you should manually create the
  'install-clang-apply-replacements-stripped' target.
Call Stack (most recent call first):
  CMakeLists.txt:1367 (llvm_distribution_add_targets)


CMake Error at cmake/modules/LLVMDistributionSupport.cmake:276 (message):
  Specified distribution component 'clang-include-fixer' doesn't have an
  install target
Call Stack (most recent call first):
  CMakeLists.txt:1367 (llvm_distribution_add_targets)


CMake Error at cmake/modules/LLVMDistributionSupport.cmake:284 (message):
  Specified distribution component 'clang-include-fixer' doesn't have an
  install-stripped target.  Its installation target creation should be
  changed to use add_llvm_install_targets, or you should manually create the
  'install-clang-include-fixer-stripped' target.
Call Stack (most recent call first):
  CMakeLists.txt:1367 (llvm_distribution_add_targets)


CMake Error at cmake/modules/LLVMDistributionSupport.cmake:276 (message):
  Specified distribution component 'clang-tidy' doesn't have an install
  target
Call Stack (most recent call first):
  CMakeLists.txt:1367 (llvm_distribution_add_targets)


CMake Error at cmake/modules/LLVMDistributionSupport.cmake:284 (message):
  Specified distribution component 'clang-tidy' doesn't have an
  install-stripped target.  Its installation target creation should be
  changed to use add_llvm_install_targets, or you should manually create the
  'install-clang-tidy-stripped' target.
Call Stack (most recent call first):
  CMakeLists.txt:1367 (llvm_distribution_add_targets)


CMake Error at cmake/modules/LLVMDistributionSupport.cmake:276 (message):
  Specified distribution component 'clangd' doesn't have an install target
Call Stack (most recent call first):
  CMakeLists.txt:1367 (llvm_distribution_add_targets)


CMake Error at cmake/modules/LLVMDistributionSupport.cmake:284 (message):
  Specified distribution component 'clangd' doesn't have an install-stripped
  target.  Its installation target creation should be changed to use
  add_llvm_install_targets, or you should manually create the
  'install-clangd-stripped' target.
Call Stack (most recent call first):
  CMakeLists.txt:1367 (llvm_distribution_add_targets)


CMake Error at cmake/modules/LLVMDistributionSupport.cmake:276 (message):
  Specified distribution component 'find-all-symbols' doesn't have an install
  target
Call Stack (most recent call first):
  CMakeLists.txt:1367 (llvm_distribution_add_targets)


CMake Error at cmake/modules/LLVMDistributionSupport.cmake:284 (message):
  Specified distribution component 'find-all-symbols' doesn't have an
  install-stripped target.  Its installation target creation should be
  changed to use add_llvm_install_targets, or you should manually create the
  'install-find-all-symbols-stripped' target.
Call Stack (most recent call first):
  CMakeLists.txt:1367 (llvm_distribution_add_targets)
  ```

Hi, I've been looking into this for the past few days. I'm not sure what I'm doing tbh but I believe I know the reason behind this compilation failure.

## Problem

As you know, building `compiler-rt`  is a prerequisite for building `all`, `clang`, `check-all`, `check-clang` etc. As outlined by this issue, while one of its steps for `builtins-mos-unknown-unknown` compiles as expected, `runtimes-mos-unknown-unknown` does not. IIRC, `runtimes-mos-unknown-unknown` is supposed to target MOS processors. Instead, it tries to target the host computer, i.e., Apple Silicon. This leads to multiple issues, e.g., not a single compiler flag such as `-g` works, which, in turn, causes compilation to fail.

## "Solution"

To fix that, I initially inserted the following line into [`clang/cmake/caches/MOS.cmake`](https://github.com/llvm-mos/llvm-mos/blob/main/clang/cmake/caches/MOS.cmake#L20):

```cmake
set(RUNTIMES_mos-unknown-unknown_CMAKE_SYSTEM_NAME Generic CACHE STRING "")
```

This should have the similar effect of the following line. These lines should force `compiler-rt` to be compiled against MOS:

https://github.com/llvm-mos/llvm-mos/blob/4286cb71c3613890f1c9c644b6ecc3772ecbed7c/clang/cmake/caches/MOS.cmake#L20

## Flaws in "Solution"

But for whatever reason, my POS computer ignores `RUNTIMES_mos-unknown-unknown_CMAKE_SYSTEM_NAME` 💀. The following CMake error log replaces the initial log by OP (btw took the liberty to add debug statements to CMake scripts. E.g., a few statements confirm that the `APPLE` CMake variable is set):

```
[proc] Executing command: /opt/homebrew/bin/cmake --build /Users/pr1metine/projects/llvm/llvm-mos/build --config Debug --target all --
[build] [1/7   0% :: 0.416] Performing build step for 'builtins-mos-unknown-unknown'
[build] ninja: no work to do.
[build] [2/7  14% :: 0.428] No install step for 'builtins-mos-unknown-unknown'
[build] [3/7  42% :: 0.446] Completed 'builtins-mos-unknown-unknown'
[build] [4/7  42% :: 0.446] Performing configure step for 'runtimes-mos-unknown-unknown'
[...]
[build] -- CMAKE_SYSTEM_NAME: Darwin, TEST: Generic
[build] -- Could NOT find Terminfo (missing: Terminfo_LINKABLE) 
[build] -- Could NOT find Terminfo (missing: Terminfo_LINKABLE) 
[build] CMake Warning at /Users/pr1metine/projects/llvm/llvm-mos/llvm/cmake/modules/HandleLLVMOptions.cmake:351 (message):
[build]   -fPIC is not supported.
[build] Call Stack (most recent call first):
[build]   /Users/pr1metine/projects/llvm/llvm-mos/llvm/cmake/modules/HandleLLVMOptions.cmake:411 (add_flag_or_print_warning)
[build]   CMakeLists.txt:167 (include)
[build] 
[build] 
[build] -- LLVM host triple: arm64-apple-darwin23.4.0
[build] -- LLVM default target triple: mos-unknown-unknown
[build] -- Found libtool - /Users/pr1metine/projects/llvm/llvm-mos/build/bin/llvm-libtool-darwin
[...]
[build] -- Found OSX_SYSROOT: /Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX14.5.sdk
[build] -- Found libtool - /Users/pr1metine/projects/llvm/llvm-mos/build/bin/llvm-libtool-darwin
[build] -- cmake c compiler target: mos-unknown-unknown
[build] -- APPLE STATUS: 1
[build] -- CMAKE_SYSTEM_NAME: Darwin
[build] -- CMAKE_SYSTEM_NAME?: 
[build] -- APPLE IS SET INDEED
[build] -- Got ld supported ARCHES: armv6 armv7 armv7s arm64 arm64e arm64_32 i386 x86_64 x86_64h armv6m armv7k armv7m armv7em (tvOS)
[build] -- Toolchain supported arches: armv6;armv7;armv7s;arm64;arm64e;arm64_32;i386;x86_64;x86_64h;armv6m;armv7k;armv7m;armv7em;(tvOS)
[build] -- Finding valid architectures for osx...
[build] -- Disabling i386 slice for DARWIN_osx_ARCHS
[build] -- OSX supported arches: 
[build] -- Finding valid architectures for iossim...
[build] -- Disabling i386 slice for iossim
[build] -- ios Simulator supported arches: 
[build] -- Finding valid architectures for ios...
[build] -- Disabling sanitizers armv7* slice for ios
[build] -- ios supported arches: 
[build] -- Compiler-RT supported architectures: 
[build] -- Host linker version: 530
[build] CMake Error at /Users/pr1metine/projects/llvm/llvm-mos/compiler-rt/CMakeLists.txt:476 (message):
[build]   -g is not supported by host compiler
[build] 
[build] 
[build] -- Configuring incomplete, errors occurred!
[build] FAILED: runtimes/runtimes-mos-unknown-unknown-stamps/runtimes-mos-unknown-unknown-configure /Users/pr1metine/projects/llvm/llvm-mos/build/runtimes/runtimes-mos-unknown-unknown-stamps/runtimes-mos-unknown-unknown-configure 
[build] cd /Users/pr1metine/projects/llvm/llvm-mos/build/runtimes/runtimes-mos-unknown-unknown-bins && /opt/homebrew/Cellar/cmake/3.29.3/bin/cmake --no-warn-unused-cli -DCMAKE_C_COMPILER=/Users/pr1metine/projects/llvm/llvm-mos/build/./bin/clang -DCMAKE_CXX_COMPILER=/Users/pr1metine/projects/llvm/llvm-mos/build/./bin/clang++ -DCMAKE_ASM_COMPILER=/Users/pr1metine/projects/llvm/llvm-mos/build/./bin/clang -DCMAKE_LINKER=/Users/pr1metine/projects/llvm/llvm-mos/build/./bin/ld.lld -DCMAKE_AR=/Users/pr1metine/projects/llvm/llvm-mos/build/./bin/llvm-ar -DCMAKE_RANLIB=/Users/pr1metine/projects/llvm/llvm-mos/build/./bin/llvm-ranlib -DCMAKE_NM=/Users/pr1metine/projects/llvm/llvm-mos/build/./bin/llvm-nm -DCMAKE_OBJDUMP=/Users/pr1metine/projects/llvm/llvm-mos/build/./bin/llvm-objdump -DCMAKE_OBJCOPY=/Users/pr1metine/projects/llvm/llvm-mos/build/./bin/llvm-objcopy -DCMAKE_STRIP=/Users/pr1metine/projects/llvm/llvm-mos/build/./bin/llvm-strip -DCMAKE_READELF=/Users/pr1metine/projects/llvm/llvm-mos/build/./bin/llvm-readelf -DCMAKE_C_COMPILER_TARGET=mos-unknown-unknown -DCMAKE_CXX_COMPILER_TARGET=mos-unknown-unknown -DCMAKE_ASM_COMPILER_TARGET=mos-unknown-unknown -DCMAKE_INSTALL_PREFIX=/usr/local -DLLVM_BINARY_DIR=/Users/pr1metine/projects/llvm/llvm-mos/build -DLLVM_CONFIG_PATH=/Users/pr1metine/projects/llvm/llvm-mos/build/bin/llvm-config -DLLVM_ENABLE_WERROR=OFF -DLLVM_HOST_TRIPLE=arm64-apple-darwin23.4.0 -DLLVM_HAVE_LINK_VERSION_SCRIPT=0 -DLLVM_USE_RELATIVE_PATHS_IN_DEBUG_INFO=OFF -DLLVM_USE_RELATIVE_PATHS_IN_FILES=OFF -DLLVM_LIT_ARGS=-sv -DLLVM_SOURCE_PREFIX= -DPACKAGE_VERSION=19.0.0git -DCMAKE_BUILD_TYPE=Debug -DCMAKE_MAKE_PROGRAM=/opt/homebrew/bin/ninja -DCMAKE_EXPORT_COMPILE_COMMANDS=1 -DCOMPILER_RT_BUILD_BUILTINS=OFF -DLLVM_INCLUDE_TESTS=ON -DLLVM_ENABLE_PROJECTS_USED=ON -DLLVM_ENABLE_PER_TARGET_RUNTIME_DIR=OFF -DCMAKE_C_COMPILER_WORKS=ON -DCMAKE_CXX_COMPILER_WORKS=ON -DCMAKE_ASM_COMPILER_WORKS=ON -DCOMPILER_RT_DEFAULT_TARGET_ONLY=ON -DLLVM_RUNTIMES_TARGET=mos-unknown-unknown -DHAVE_LLVM_LIT=ON -DCLANG_RESOURCE_DIR= -DLLVM_DEFAULT_TARGET_TRIPLE=mos-unknown-unknown -DLLVM_ENABLE_RUNTIMES=compiler-rt -DCMAKE_SYSTEM_NAME=Generic -DTEST=Generic -GNinja -S /Users/pr1metine/projects/llvm/llvm-mos/llvm/runtimes/../../runtimes -B /Users/pr1metine/projects/llvm/llvm-mos/build/runtimes/runtimes-mos-unknown-unknown-bins && /opt/homebrew/Cellar/cmake/3.29.3/bin/cmake -E touch /Users/pr1metine/projects/llvm/llvm-mos/build/runtimes/runtimes-mos-unknown-unknown-stamps/runtimes-mos-unknown-unknown-configure
[build] ninja: build stopped: subcommand failed.
[proc] The command: /opt/homebrew/bin/cmake --build /Users/pr1metine/projects/llvm/llvm-mos/build --config Debug --target all -- exited with code: 1
[driver] Build completed: 00:00:11.856
[build] Build finished with exit code 1
```

Note that the CMake configure command for `runtimes-mos-unknown-unknown` lists `-DCMAKE_SYSTEM_NAME=Generic`.

## Workaround

As a temporary workaround, I added a few lines in [`runtimes/CMakeLists.txt`](https://github.com/llvm-mos/llvm-mos/blob/main/runtimes/CMakeLists.txt):

```cmake
message(STATUS "CMAKE_SYSTEM_NAME: ${CMAKE_SYSTEM_NAME}, TEST: ${TEST}")
set(CMAKE_SYSTEM_NAME "Generic")
unset(APPLE)
```

With this, `clang` and `all` successfully compile. This is the log produced by CMake:

```cmake
[main] Building folder: llvm-mos 
[build] Starting build
[proc] Executing command: /opt/homebrew/bin/cmake --build /Users/pr1metine/projects/llvm/llvm-mos/build --config Debug --target all --
[build] [1/7   0% :: 0.479] Performing build step for 'builtins-mos-unknown-unknown'
[build] ninja: no work to do.
[build] [2/7  14% :: 0.492] No install step for 'builtins-mos-unknown-unknown'
[build] [3/7  42% :: 0.509] Completed 'builtins-mos-unknown-unknown'
[build] [4/7  42% :: 0.509] Performing configure step for 'runtimes-mos-unknown-unknown'
[...]
[build] Not searching for unused variables given on the command line.
[build] CMake Warning at /opt/homebrew/Cellar/cmake/3.29.3/share/cmake/Modules/Platform/Darwin-Initialize.cmake:308 (message):
[build]   Ignoring CMAKE_OSX_SYSROOT value:
[build] 
[build]    /Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX14.4.sdk
[build] 
[build]   because the directory does not exist.
[build] Call Stack (most recent call first):
[build]   /opt/homebrew/Cellar/cmake/3.29.3/share/cmake/Modules/CMakeSystemSpecificInitialize.cmake:34 (include)
[build]   CMakeLists.txt:11 (project)
[build] 
[build] 
[build] -- CMAKE_SYSTEM_NAME: Darwin, TEST: Generic
[build] -- Could NOT find Terminfo (missing: Terminfo_LINKABLE) 
[build] -- Could NOT find Terminfo (missing: Terminfo_LINKABLE) 
[build] CMake Warning at /Users/pr1metine/projects/llvm/llvm-mos/llvm/cmake/modules/HandleLLVMOptions.cmake:351 (message):
[build]   -fPIC is not supported.
[build] Call Stack (most recent call first):
[build]   /Users/pr1metine/projects/llvm/llvm-mos/llvm/cmake/modules/HandleLLVMOptions.cmake:411 (add_flag_or_print_warning)
[build]   CMakeLists.txt:167 (include)
[build] 
[build] 
[build] -- LLVM host triple: arm64-apple-darwin23.4.0
[build] -- LLVM default target triple: mos-unknown-unknown
[...]
[build] -- cmake c compiler target: mos-unknown-unknown
[build] -- APPLE STATUS: 
[build] -- CMAKE_SYSTEM_NAME: Generic
[build] -- CMAKE_SYSTEM_NAME?: 
[build] -- Compiler-RT supported architectures: mos
[build] -- check-shadowcallstack does nothing.
[build] -- check-compiler-rt does nothing.
[build] -- check-runtimes does nothing.
[build] -- Configuring done (0.5s)
[build] -- Generating done (0.0s)
[build] -- Build files have been written to: /Users/pr1metine/projects/llvm/llvm-mos/build/runtimes/runtimes-mos-unknown-unknown-bins
[build] [5/7  57% :: 1.065] Performing build step for 'runtimes-mos-unknown-unknown'
[build] ninja: no work to do.
[build] [6/7  71% :: 1.078] No install step for 'runtimes-mos-unknown-unknown'
[build] [7/7 100% :: 1.096] Completed 'runtimes-mos-unknown-unknown'
[driver] Build completed: 00:00:01.112
[build] Build finished with exit code 0
```

## Problems with Workaround

However, `check-clang` and `check-all` fail. The following log shows an except of the errors occurring during `check-clang`. Out of all clang tests, 638 or 1.76 % of all tests failed:

https://gist.github.com/pr1metine/7df71de72b7dc8413f945960c68e5bd1

## Call to action

I'm running out of ideas rn so to anyone reading this (looking at you Daniel):

- From what I can tell from the CI logs, `runtimes-mos-unknown-unknown` is not built by the existing GitHub Actions hosts, i.e., Windows, Ubuntu, and Intel Macs. What's the purpose of this? Do we even need to compile this?
- I have no clue why tf `cmake -DCMAKE_SYSTEM_NAME=Generic [...]` does not set `CMAKE_SYSTEM_NAME` (See "Problems of 'Solution'"). What am I doing wrong?
- I am also clueless about the error logs produced when targeting `check-clang`. What's going wrong there?

I will keep working on this for a few days. Would love to get some feedback on this.

--- a/.github/workflows/llvm-project-tests.yml
+++ b/.github/workflows/llvm-project-tests.yml
@@ -14,7 +14,7 @@ on:
         required: false
       os_list:
         required: false
-        default: '["ubuntu-latest", "windows-2019", "macOS-13"]'
+        default: '["ubuntu-latest", "windows-2019", "macOS-13", "macos-14"]'
       python_version:
         required: false
         type: string

--- a/clang/cmake/caches/MOS.cmake
+++ b/clang/cmake/caches/MOS.cmake
@@ -18,6 +18,7 @@ set(BUILTINS_mos-unknown-unknown_COMPILER_RT_BAREMETAL_BUILD ON CACHE BOOL "")
 set(BUILTINS_mos-unknown-unknown_COMPILER_RT_BUILTINS_ENABLE_PIC OFF CACHE BOOL "")
 set(BUILTINS_mos-unknown-unknown_CMAKE_BUILD_TYPE MinSizeRel CACHE BOOL "")
 set(BUILTINS_mos-unknown-unknown_CMAKE_SYSTEM_NAME Generic CACHE STRING "")
+set(RUNTIMES_mos-unknown-unknown_CMAKE_SYSTEM_NAME Generic CACHE STRING "")
 
 set(LLVM_DEFAULT_TARGET_TRIPLE "mos-unknown-unknown" CACHE STRING "")
 

